js/controllers.js

```diff
@@ -1,12 +1,192 @@
+
+
+chartControllers.controller('chartCtrl', ['$scope', '$location', 'Datax',
+    function($scope, $location, Datax) {
+        
+
+
+
+        //csv file data is load for all charts libraries
+        $scope.loadData = function(fileName) {   
+            //console.log("loading data "+fileName);
+            data = getData(fileName);
+            $scope.setActive('b1',fileName);
+            //$scope.Datax = Datax;
+            console.log(Datax)
+            loadYAxisLabel(fileName);
+            $scope.a = true;
+            drawChart();
+        };
+
+                //default one 
+        var myChart = {};
+       
+        $scope.loadChartLibrary = function(libraryIndex) {   
+            myChart = libraryName[libraryIndex];            
+            switchChartLib(myChart.library);
+            $scope.setActive('b2',myChart.library);
+            $scope.b = true;
+            drawChart();
+            
+        };
+ 
+
+        //getDataing highchart sample 
+        $scope.chartConfig = highchart.chartConfig;
+
+        //getting nvd3 chart sample
+        $scope.options = ncd3Chart1.options;
+        $scope.data = ncd3Chart1.data;
+
+        //getting google chart sample
+        $scope.chart = googleChart;
+
+        //Switch all chart libaraies chart types at one
+        $scope.switchChartType = function(gchartType, hchartType, dchartType) {
+            $scope.chart.type = gchartType;
+            $scope.chartConfig.options.chart.type = hchartType;
+            $scope.options.chart.type = dchartType;
+            $scope.setActive('b3',hchartType);
+            //nvd3             
+            loadYAxisLabel($scope.fileName);
+            var nvdLabels = nvd3AxisLabels;
+            $scope.options.chart.xAxis = {
+                'axisLabel': 'Make'
+            };
+            $scope.options.chart.xAxis = {
+                'tickFormat': function(d) {
+                    return nvdLabels[d]
+                }
+            };
+
+        };
+
+        //deafult button picker
+        $scope.active={'b1':false,'b2':false,'b3':'bar'};
+
+        $scope.setActive = function(set,type) {
+            if(set=='b1')
+                $scope.active.b1 = type;
+            if(set=='b2')
+                $scope.active.b2 = type;
+            if(set=='b3')
+                $scope.active.b3 = type;
+        };
+
+        $scope.isBActive = function(set,type) {
+            if(set=='b1')
+              return type === $scope.active.b1;
+            if(set=='b2')
+                return type === $scope.active.b2;
+            if(set=='b3')
+                return type === $scope.active.b3;
+        };
+        
+        switchChartLib = function(chartLib) {
+            console.log('switchChartLib to '+chartLib);
+            $scope.highxChart = false;
+            $scope.googlexChart = false;
+            $scope.NVD3Chart = false;
+            $scope[chartLib] = true;
+        };

```

Comment:
you should consider something like (pseudo code here) `$scope[chartLib] = {highx | googlex | nvd3}` instead of having three boolean properties on your object.